packages/hash/api/src/graph/ontology/primitive/data-type.ts

```diff
@@ -0,0 +1,121 @@
+import { AxiosError } from "axios";
+import { generateTypeId } from "@hashintel/hash-shared/ontology-types";
+import { DataType } from "@blockprotocol/type-system";
+import { DataTypeWithMetadata, VersionedUri } from "@hashintel/hash-subgraph";
+import { versionedUriFromComponents } from "@hashintel/hash-subgraph/src/shared/type-system-patch";
+import { getNamespaceOfAccountOwner } from "./util";
+import { GraphContext } from "../..";
+
+/**
+ * Create a data type.
+ *
+ * @todo revisit data type creation
+ *   User defined data types are not specified yet, which means this `create`
+ *   operation should not be exposed to users yet.
+ *   Depends on the RFC captured by:
+ *   https://app.asana.com/0/1200211978612931/1202464168422955/f
+ *
+ * @param params.ownedById - the id of the account who owns the data type
+ * @param params.schema - the `DataType`
+ * @param params.actorId - the id of the account that is creating the data type
+ */
+export const createDataType = async (
+  { graphApi }: GraphContext,
+  params: {
+    ownedById: string;
```

Comment:
We can make use of the type aliases already from here: https://github.com/hashintel/hash/blob/main/packages/hash/shared/src/types.ts#L4
I know we have an `OwnedById` and `UpdatedById` type here, but I think these params should almost all be `AccountId` to support downcasting any kind of user identifier.

This change wouldn't break any code for you right now

Reply:
Ah, apologies I forgot about this - will add these now